packages/flutter_tools/lib/src/base/deferred_component.dart

```diff
@@ -0,0 +1,216 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'package:meta/meta.dart';
+
+import '../base/file_system.dart';
+import '../convert.dart';
+import '../globals.dart' as globals;
```

Comment:
Don't introduce new usages of globals. Have parseLoadingUnitManifest accept a logger as an argument if you need one